test/extensions/filters/listener/common/fuzz/listener_filter_fuzzer.h

```diff
@@ -29,23 +31,35 @@ class ListenerFilterFuzzer {
   FakeConnectionSocket socket_;
   NiceMock<Event::MockDispatcher> dispatcher_;
   Event::FileReadyCb file_event_callback_;
+  uint32_t events_;
+  envoy::config::core::v3::Metadata metadata_;
 };
 
-class FuzzedHeader {
+class FuzzedInputStream {
 public:
-  FuzzedHeader(const test::extensions::filters::listener::FilterFuzzTestCase& input);
+  FuzzedInputStream(const test::extensions::filters::listener::FilterFuzzTestCase& input);
 
-  // Copies next read into buffer and returns the number of bytes written
-  Api::SysCallSizeResult next(void* buffer, size_t length);
+  FuzzedInputStream(const std::vector<uint8_t> buffer, const std::vector<size_t> indices);
```

Comment:
Having `const` vector inputs gives you the worst of both worlds: callers can't use `std::move` to avoid making a copy when constructing the argument, and the function body can't use `std::move` to move the argument. Check out https://abseil.io/tips/117.